src/sentry/static/sentry/app/components/onboardingWizard/toDoItem.jsx

```diff
@@ -0,0 +1,135 @@
+import moment from 'moment';
+import PropTypes from 'prop-types';
+import React from 'react';
+import {t, tct} from '../../locale';
+
+import OrganizationState from '../../mixins/organizationState';
+import Confirmation from './confirmation';
+
+const TodoItem = React.createClass({
+  propTypes: {
+    task: PropTypes.object,
+    onSkip: PropTypes.func.isRequired
+  },
+
+  mixins: [OrganizationState],
+
+  getInitialState: function() {
+    return {
+      showConfirmation: false,
+      isExpanded: false
+    };
+  },
+  toggleDescription(e) {
+    this.setState({isExpanded: !this.state.isExpanded});
+  },
+
+  toggleConfirmation: function() {
+    this.setState({showConfirmation: !this.state.showConfirmation});
+  },
+
+  formatDescription: function() {
+    if (this.state.isExpanded) {
+      return <p>{this.props.task.description}. {this.props.task.detailedDescription}</p>;
+    } else {
+      return <p>{this.props.task.description}</p>;
+    }
+  },
+
+  learnMoreUrlCreator: function(org) {
+    let learnMoreUrl;
+    if (this.props.task.featureLocation === 'project') {
+      learnMoreUrl =
+        '/organizations/' +
+        org.slug +
+        '/projects/choose/?onboarding=1&task=' +
+        this.props.task.task;
+    } else if (this.props.task.featureLocation === 'organization') {
+      learnMoreUrl = '/organizations/' + org.slug + '/' + this.props.task.location;
+    } else if (this.props.task.featureLocation === 'absolute') {
+      learnMoreUrl = this.props.task.location;
+    }
+    return learnMoreUrl;
+  },
+
+  skip: function(task) {
+    this.props.onSkip(task);
+    this.setState({showConfirmation: false});
+  },
+
+  render: function() {
+    let org = this.getOrganization();
+    let learnMoreUrl = this.learnMoreUrlCreator(org);
+    let classNames = '';
+    let description;
+
+    switch (this.props.task.status) {
+      case 'complete':
+        classNames += ' checked';
+        description = tct('[user] completed [dateCompleted]', {
+          user: this.props.task.user,
+          dateCompleted: moment(this.props.task.dateCompleted).fromNow()
+        });
+        break;
+      case 'pending':
+        classNames += ' pending';
+        description = tct('[user] kicked off [dateCompleted]', {
+          user: this.props.task.user,
+          dateCompleted: moment(this.props.task.dateCompleted).fromNow()
+        });
+        break;
+      case 'skipped':
+        classNames += ' skipped';
+        description = tct('[user] skipped [dateCompleted]', {
+          user: this.props.task.user,
+          dateCompleted: moment(this.props.task.dateCompleted).fromNow()
+        });
+        break;
+      default:
+        description = this.formatDescription();
+    }
+
+    if (this.state.showConfirmation) {
+      classNames += ' blur';
```

Comment:
the classnames library would make this logic simpler, lmk if you want an example